react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      openId: 0
+    };
+    this.counting();
+  }
+
+  counting() {
+    this.timerID = setInterval(
+      () => this.tick(), 3000);
+  }
+
+  tick() {
+    if (this.state.openId === 4) {
+      this.setState({ openId: 0 });
+    } else {
+      this.setState({ openId: this.state.openId + 1 });
+    }
+  }
+
+  handleClickRight() {
+    if (this.state.openId === 4) {
```

Comment:
Same here, we want to dynamically set this value, so if the array size changes it will still work.